### PR TITLE
BM40 keymap/dan layout update

### PR DESCRIPTION
**Pull request by syntax-magic**
BM40 Keyboard layout update for keymaps/dan.

## Description

   - Moved RGB keys to RightFN layer
   - Added mouse keys in MOU layer
   - Removed keys that use tap dance

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/kprepublic/bm40hsrgb/keymaps/dan/rules.mk [hunk 1]
```diff
@@ -1,2 +1,4 @@
-TAP_DANCE_ENABLE = yes
 SRC += features/custom_shift_keys.c
+TAP_DANCE_ENABLE = no
+IGNORE_MOD_TAP_INTERRUPT = yes
+HOLD_ON_OTHER_KEY_PRESS = no
```
Comment (zvecr) [inline keyboards/kprepublic/bm40hsrgb/keymaps/dan/rules.mk:4]: These are not `rules.mk` settings, so specifying them here will do absolutely nothing.
```suggestion
```

File: keyboards/kprepublic/bm40hsrgb/keymaps/dan/rules.mk [hunk 2]
```diff
@@ -1,2 +1,4 @@
-TAP_DANCE_ENABLE = yes
 SRC += features/custom_shift_keys.c
+TAP_DANCE_ENABLE = no
```
Comment (zvecr) [inline keyboards/kprepublic/bm40hsrgb/keymaps/dan/rules.mk:2]: Setting this to "no" is the default, so also does nothing
```suggestion
```

File: keyboards/kprepublic/bm40hsrgb/keymaps/dan/keymap.c [hunk 3]
```diff
@@ -18,56 +18,69 @@
...
+// enum {
+//     TD_PAST_PSLS,
+//     TD_PMNS_PPLS,
+// };
 
-tap_dance_action_t tap_dance_actions[] = {
-    [TD_PAST_PSLS]  = ACTION_TAP_DANCE_DOUBLE(KC_PAST,    KC_PSLS),
-    [TD_PASTE_COPY] = ACTION_TAP_DANCE_DOUBLE(LCTL(KC_V), LCTL(KC_C))
-};
+// tap_dance_action_t tap_dance_actions[] = {
```
Comment (zvecr) [inline keyboards/kprepublic/bm40hsrgb/keymaps/dan/keymap.c:51]: We also do not want commented out code in the repo, which there is a fair amount introduced in this file.

**Review by zvecr — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Comment by zvecr** [comment]
Unfortunately QMK has [changed its stance](https://docs.qmk.fm/#/ChangeLog/20230827?id=notable-changes) on accepting user keymaps to the repository.

> qmk/qmk_firmware is no longer accepting PRs for keymaps other than for manufacturer-supported keymaps. User keymap workflow has been documented [here](https://docs.qmk.fm/#/newbs) for several years. This change is to progressively reduce the maintenance burden on the project, and to allow us to focus on the core features of QMK.

QMK Firmware now officially supports [storing user keymaps outside of the normal QMK Firmware repository](https://docs.qmk.fm/#/newbs_external_userspace), allowing users to maintain their own keymaps without having to fork, modify, and maintain a copy of QMK Firmware themselves. 

While we appreciate the time that was taken on this PR, hopefully you understand we are closing it, so that QMK can improve in the long term.

**Pull request closed without merging**